pkg/wallet/api.go

```diff
@@ -55,49 +56,24 @@ type Crypto interface {
 	// error: error
 	SignMessage(message []byte, fromVerKey string) ([]byte, error)
 
-	// DecryptMessage decrypt message
-	//
-	// Args:
-	//
-	// encMessage: The encrypted message content
+	// DeriveKEK will derive an ephemeral symmetric key (kek) using a private from key fetched from
+	// from the wallet corresponding to fromPubKey and derived with toPubKey.
 	//
-	// toVerKey:The verification key of the recipient.
+	// This function assumes both fromPubKey and toPubKey to be on curve25519.
 	//
-	// []byte: Decrypted message content
-	//
-	// string: The sender verification key
-	//
-	// error: error
-	DecryptMessage(encMessage []byte, toVerKey string) ([]byte, string, error)
-}
+	// returns:
+	// 		kek []byte the key encryption key used to decrypt a cek (a shared key)
+	//		error in case of errors
+	DeriveKEK(alg, apu []byte, fromPuKey, toPubKey *[chacha.KeySize]byte) ([]byte, error)
```

Comment:
why are we setting the size of the key in this interface. this was previously fixed.

Reply:
Key derivation requires the key size used with the curv, which in our case is curve25519 and the key size is fixed..

I can hide the size from the function signature and enforce it in the implementation.

Reply:
Well. I don’t really understand why it is hidden in some methods and not others.

Reply:
Wallet.DeriveKEK now uses []byte for keys, no more key size length enforced...

Reply:
done